src/tasks/beacon/beacon_task.c

```diff
@@ -1,146 +1,87 @@
-{
-    size_t pkt_len = 0;
-    memset(data, 0, sizeof(data));
-
-    // [64] Write the current state name
-    pkt_len = write_string(data, pkt_len, slate->current_state->name);
-
-    // [64 + 4 = 68] Write the reboot counter
-    pkt_len = write_uint32(data, pkt_len, slate->reboot_counter);
-
-    // [68 + 8 = 76] Write the current time
-    pkt_len = write_uint64(data, pkt_len, slate->time_in_current_state_ms);
-
-    // [76 + 4 = 80] Write the current rx_bytes
-    pkt_len = write_uint32(data, pkt_len, slate->rx_bytes);
-
-    // [80 + 4 = 84] Write the current rx_packets
-    pkt_len = write_uint32(data, pkt_len, slate->rx_packets);
-
-    // [84 + 4 = 88] Write the current rx_backpressure_drops
-    pkt_len = write_uint32(data, pkt_len, slate->rx_backpressure_drops);
-
-    // [88 + 4 = 92] Write the current rx_bad_packet_drops
-    pkt_len = write_uint32(data, pkt_len, slate->rx_bad_packet_drops);
-
-    // [92 + 4 = 96] Write the current tx_bytes
-    pkt_len = write_uint32(data, pkt_len, slate->tx_bytes);
-
-    // [96 + 4 = 100] Write the current tx_packets
-    pkt_len = write_uint32(data, pkt_len, slate->tx_packets);
-
-    if (pkt_len > MAX_DATA_SIZE)
-    {
-        LOG_ERROR("Serialized data too long: %d", pkt_len);
-        return 0;
-    }
-
-    return pkt_len;
-}
-
-void beacon_task_init(slate_t *slate)
-{
-    memset(tmp_data, 0, sizeof(tmp_data));
-}
-
-void beacon_task_dispatch(slate_t *slate)
-{
-    // Create a new packet for radio TX
-    packet_t pkt;
-    pkt.src = 0;
-    pkt.dst = 0;
-    pkt.flags = 0;
-    pkt.seq = 0;
-
-    // Serialize data from slate variables
-    // Current state name
-    size_t pkt_len = serialize_slate(slate, tmp_data);
-
-    // Commit into serialized byte array
-    pkt.len = pkt_len;
-    memcpy(pkt.data, tmp_data, pkt.len);
-
-    // Write into tx_queue
-    if (queue_try_add(&slate->tx_queue, &pkt))
-    {
-        LOG_INFO("Beacon pkt added to queue");
-    }
-    else
-    {
-        LOG_ERROR("Beacon pkt failed to commit to tx_queue");
-    }
-}
-
-sched_task_t beacon_task = {.name = "beacon",
-                            .dispatch_period_ms = 30000,
-                            .task_init = &beacon_task_init,
-                            .task_dispatch = &beacon_task_dispatch,
-
-                            /* Set to an actual value on init */
-                            .next_dispatch = 0};
+/**
+ * @author  Thomas Haile
+ * @date    2025-05-17
+ *
+ * Task to emit telemetry packet to the radio TX queue.
+ */
+
+#include "beacon_task.h"
+
+// Some limits on the data types
+// Total of 64 chars reserved for name field.
+#define MAX_STR_LEN 64
+#define MAX_DATA_SIZE 252
+
+typedef struct {
+  uint32_t reboot_counter;
+  uint64_t time;
+  uint32_t rx_bytes;
+  uint32_t rx_packets;
+  uint32_t rx_backpressure_drops;
+  uint32_t rx_bad_packet_drops;
+  uint32_t tx_bytes;
+  uint32_t tx_packets;
+} __attribute__((__packed__)) beacon_stats;

```

Comment:
We can also try statically asserting sizeof(beacon_stats) doesn't exceed some length value (compile-time check), i.e. 252-64 = 188.